.github/workflows/semantic.yaml

```diff
@@ -1,8 +1,9 @@
-name:  Publish
+name:  Semantic-Release
 on: [pull_request, push]
 jobs:
   publish:
     runs-on: ubuntu-latest
+    permissions: write-all
```

Comment:
this seems have security risk, please check the following(and not only) docs/posts to make sure there is no such risk:
https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs#overview
https://securitylab.github.com/research/github-actions-preventing-pwn-requests/